geo/src/algorithm/coordinate_position.rs

```diff
@@ -0,0 +1,749 @@
+use crate::algorithm::{
+    bounding_rect::BoundingRect, dimensions::HasDimensions, intersects::Intersects,
+    kernels::HasKernel,
+};
+use crate::{
+    Coordinate, Geometry, GeometryCollection, Line, LineString, MultiLineString, MultiPoint,
+    MultiPolygon, Point, Polygon, Rect, Triangle,
+};
+
+/// The position of a `Coordinate` relative to a `Geometry`
+#[derive(PartialEq, Clone, Debug)]
+pub enum CoordPos {
+    OnBoundary,
+    Inside,
+    Outside,
+}
+
+/// Determine the position of a `Coordinate` relative to a geometry.
+///
+/// # Examples
+///
+/// ```rust
+/// use geo::{polygon, Coordinate};
+/// use geo::coordinate_position::{CoordinatePosition, CoordPos};
+///
+/// let square_poly = polygon![(x: 0.0, y: 0.0), (x: 2.0, y: 0.0), (x: 2.0, y: 2.0), (x: 0.0, y: 2.0), (x: 0.0, y: 0.0)];
+///
+/// let inside_coord = Coordinate { x: 1.0, y: 1.0 };
+/// assert_eq!(square_poly.coordinate_position(&inside_coord), CoordPos::Inside);
+///
+/// let boundary_coord = Coordinate { x: 0.0, y: 1.0 };
+/// assert_eq!(square_poly.coordinate_position(&boundary_coord), CoordPos::OnBoundary);
+///
+/// let outside_coord = Coordinate { x: 5.0, y: 5.0 };
+/// assert_eq!(square_poly.coordinate_position(&outside_coord), CoordPos::Outside);
+/// ```
+pub trait CoordinatePosition<T>
+where
+    T: HasKernel,
+{
+    fn coordinate_position(&self, coord: &Coordinate<T>) -> CoordPos {
+        let mut is_inside = false;
+        let mut boundary_count = 0;
+
+        self.calculate_coordinate_position(coord, &mut is_inside, &mut boundary_count);
+
+        // SFS "mod 2" boundary rule
```

Comment:
thoughts on adding a quote here?

```suggestion
        // “The boundary of an arbitrary collection of geometries whose interiors are disjoint
        // consists of geometries drawn from the boundaries of the element geometries by
        // application of the ‘mod 2’ union rule”
        //
        // ― OpenGIS Simple Feature Access § 6.1.15.1
```